h2/src/main/org/h2/mvstore/rtree/SpatialKey.java

```diff
@@ -111,6 +117,11 @@ public boolean equals(Object other) {
         return equalsIgnoringId(o);
     }
 
+    @Override
+    public int compareTypeSafe(Value v, CompareMode mode, CastDataProvider provider) {
+        return 0;
```

Comment:
Maybe throw an exception from this method? These keys aren't actually comparable.